### DISCUSSION TITLE
Autocomplete not working, fails to recognize completion sources

### DISCUSSION THREAD

**Discussion by Ryan-Kalinowski** [hrsh7th/nvim-cmp#541] [Q&A] [answered]
Hi folks, I recently attempted to integrate this plugin with built-in the lsp client to provide a more fully-featured completion experience. Unfortunately I wasn't able to get it to work at all. Autocomplete does not appear as I type and it fails to recognize any completion sources I install. It didn't seem appropriate to open an Issue since I'm not sure if this is a config issue on my end and I didn't see this Discussions space, so I [posted on stackexchange](https://vi.stackexchange.com/questions/35000/completion-sources-not-recognized-by-nvim-cmp).

I have installed the following relevant plugins using pathogen:

- cmp-nvim-lsp
- cmp-vsnip
- nvim-cmp
- nvim-lspconfig
- vim-vsnip
- vim-vsnip-integ

Here is the output from running CmpStatus:

```
# unknown source names
- nvim_lsp
```

Here is my config:

```
syntax on

filetype plugin indent on

set runtimepath=$HOME/.config/vim,$VIMRUNTIME

execute pathogen#infect()
execute pathogen#helptags()

map / <leader>

if has('nvim')
lua << EOF

vim.o.completeopt = 'menu,menuone,noselect'

local nvim_lsp = require('lspconfig')

local cmp = require('cmp')

cmp.setup({
        snippet = {
                expand = function(args)
                        vim.fn['vsnip#anonymous'](args.body)
                end,
        },
        mapping = {
                --['<C-d>'] = cmp.mapping(cmp.mapping.scroll_docs(-4), { 'i', 'c' }),
                --['<C-f>'] = cmp.mapping(cmp.mapping.scroll_docs(4), { 'i', 'c' }),
                --['<C-Space>'] = cmp.mapping(cmp.mapping.complete(), { 'i', 'c' }),
                --['<C-y>'] = cmp.config.disable, -- Specify `cmp.config.disable` if you want to remove the default `<C-y>` mapping.
                --['<C-e>'] = cmp.mapping({
                --      i = cmp.mapping.abort(),
                --      c = cmp.mapping.close(),
                --}),
                --['<CR>'] = cmp.mapping.confirm({ select = true }),
        },
        sources = cmp.config.sources({
                { name = 'nvim_lsp' },
        }),
})

local capabilities = vim.lsp.protocol.make_client_capabilities()
capabilities = require('cmp_nvim_lsp').update_capabilities(capabilities)

local servers = { 'metals', 'tsserver', 'rls', 'vimls' }
for _, lsp in ipairs(servers) do
        nvim_lsp[lsp].setup {
                capabilities = capabilities,
        }
end

EOF
endif
```

Output from `nvim --version`:
```
:$ nvim --version
NVIM v0.5.0
Build type: Release
LuaJIT 2.1.0-beta3
Compilation: clang -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -O2 -DNDEBUG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/tmp/neovim-20210702-21950-fs4dxz/neovim-0.5.0/build/config -I/tmp/neovim-20210702-21950-fs4dxz/neovim-0.5.0/src -I/usr/local/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include -I/usr/local/opt/gettext/include -I/tmp/neovim-20210702-21950-fs4dxz/neovim-0.5.0/build/src/nvim/auto -I/tmp/neovim-20210702-21950-fs4dxz/neovim-0.5.0/build/include
Compiled by brew@iMac-Pro

Features: +acl +iconv +tui
```

I'm hoping somebody here can suggest methods to debug or get more information about what's going wrong.

**Comment by Shougo** [comment]
You should upload the minimal config.

**Reply by Ryan-Kalinowski** [reply]
I edited to include the config.

**Comment by hrsh7th** [comment]
First, Is your language server working? (For example, diagnostics/goto_definition/formatting etc)

If you can't get only completion worked, it seems nvim-cmp's bug or somthing.

**Reply by Ryan-Kalinowski** [reply]
Built-in lsp client functionality is still working. Can you give any advice on how to get more output about what's going wrong with autocomplete? Undocumented commands, debug logging, etc?

**Comment by mjlbach** [comment]
Did you check the examples on the lspconfig wiki? They are extremely explicit

**Reply by Ryan-Kalinowski** [reply]
Yes I tried replacing my config with [this example](https://github.com/neovim/nvim-lspconfig/wiki/Autocompletion#nvim-cmp), modified to use vsnip instead of luasnip. The problem remained.

**Reply by mjlbach** [reply]
Have you tried using the *exact* code to see if it works?  The containerized version should be bullet proof. https://github.com/nvim-lsp/try.nvim

**Comment by Ryan-Kalinowski** [comment, marked as answer]
This was an issue with the completion source repository structure. Pathogen isn't able to load them correctly. I switched to packer and everything is working correctly.
